srm.js

```diff
@@ -110,8 +117,22 @@ class CodeDxApiClient {
         }
     }
 
-    async runAnalysis(projectId, formData) {
-        const response = await this.http.post(`/api/projects/${projectId}/analysis`, formData, { headers: formData.getHeaders() }).catch(rethrowError)
+    async getProjectBranches(projectId) {
+        const branchesResponse = await this.http.get(`/x/projects/${projectId}/branches`).catch(rethrowError)
+        return branchesResponse.data
+    }
+
+    async runAnalysis(projectId, baseBranchName, targetBranchName, formData) {
+        const baseBranchPresent = !!baseBranchName
+        const targetBranchPresent = !!targetBranchName
+
+        // If base branch is not defined, OR, base branch is present but target branch isn't, add the base branch name in the project URL part
+        const projectUrlPart = ((!baseBranchName) || (baseBranchPresent && targetBranchPresent)) ? "" : `;branch=${baseBranchName}`
+
+        // If target branch is not defined, no need to add the branch name to the query param. It will work on the default branch
+        const queryParamsPart = (!targetBranchName) ? "analysis" : `analysis?branchName=${targetBranchName}`
+
+        const response = await this.http.post(`/api/projects/${projectId}${projectUrlPart}/${queryParamsPart}`, formData, { headers: formData.getHeaders() }).catch(rethrowError)
```

Comment:
Still weird that part of the URL is stored in a variable called `queryParamsPart`

The logic in `projectUrlPart` is also slightly misleading due to the use of `xBranchPresent`, where the naming suggests the branch exists in SRM, but it really just represents whether the branch was specified

I suggest removing the `xBranchPresent` vars to simplify and make the actual logic clearer, and move the `analysis` URL part out of `queryParamsPart`:

```js
const projectSpecifier = baseBranchName ? `${projectId};branch=${baseBranchName}` : projectId
const queryParamsPart = targetBranchName ? `?branchName=${targetBranchName}` : ''

const response = await this.post(`/api/projects/${projectSpecifier}/analysis${queryParamsPart}`, ...)
```